util/chplenv/chpl_comm_segment.py

```diff
@@ -21,6 +21,8 @@ def get():
                 segment_val = 'large'
             else:
                 segment_val = 'everything'
+        else:
+            check_valid_var("CHPL_GASNET_SEGMENT", segment_val, ("none", "fast", "large", "everything"))
```

Comment:
Same here about inverting the conditional. 

OR make all of these call check_valid_var after setting the default, before returning. That is also reasonable.

BTW I would expect that CHPL_GASNET_SEGMENT=none is not a valid value if CHPL_COMM=gasnet. (I would expect it is used when CHPL_COMM=none for example though). Do you know differently?

Reply:
> BTW I would expect that CHPL_GASNET_SEGMENT=none is not a valid value if CHPL_COMM=gasnet. (I would expect it is used when CHPL_COMM=none for example though). Do you know differently?

I think you are right, I will remove `none` as a valid option when `CHPL_COMM=gasnet`